lib/components/ClinicUserSelect.js

```diff
@@ -0,0 +1,162 @@
+/*
+* == BSD2 LICENSE ==
+* Copyright (c) 2014, Tidepool Project
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the associated License, which is identical to the BSD 2-Clause
+* License as published by the Open Source Initiative at opensource.org.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the License for more details.
+*
+* You should have received a copy of the License along with this program; if
+* not, you can obtain one from Tidepool Project at tidepool.org.
+* == BSD2 LICENSE ==
+*/
+
+var _ = require('lodash');
+var React = require('react');
+var Select = require('react-select');
+var sundial = require('sundial');
+var cx = require('classnames');
+
+var styles = require('../../styles/components/ClinicUserSelect.module.less');
+ 
+var ClinicUserSelect = React.createClass({
+  propTypes: {
+    allUsers: React.PropTypes.object.isRequired,
+    onUserChange: React.PropTypes.func.isRequired,
+    targetId: React.PropTypes.string,
+    targetUsersForUpload: React.PropTypes.array.isRequired,
+    onAddUserClick: React.PropTypes.func.isRequired,
+    setTargetUser: React.PropTypes.func.isRequired
+  },
+
+  handleChange: function(e) {
+    this.props.setTargetUser(e);
+  },
+
+  handleClickNext: function(e) {
+    e.preventDefault();
+    if(this.props.targetId){
+      this.props.onUserChange(this.props.targetId);
+    }
+  },
+
+  valueRenderer: function(option) {
+    var user = _.get(this.props.allUsers, option.value);
+    var name = _.get(user, 'fullName');
```

Comment:
Well it's just in other places you were taking pains to account for legacy child accounts. If we need to do that (and it sounds like you believe we expect clinics to potentially still be interacting with the old accounts for a while), then we need to be consistent about it. At this point it's really ripe for factoring out something like blip's `personUtils` (I think that's what we call them) to pass in a user and get the appropriate `fullName` no matter what kind of account it is.